lang/csharp/src/apache/main/Util/Decimal.cs

```diff
@@ -0,0 +1,157 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Globalization;
+using System.Numerics;
+using Avro.Generic;
+
+namespace Avro.Util
+{
+    /// <summary>
+    /// The 'decimal' logical type.
+    /// </summary>
+    public class Decimal : LogicalType
+    {
+        /// <summary>
+        /// The logical type name for Decimal.
+        /// </summary>
+        public static readonly string LogicalTypeName = "decimal";
+
+        /// <summary>
+        /// Initializes a new Decimal logical type.
+        /// </summary>
+        public Decimal() : base(LogicalTypeName)
+        { }
+
+        /// <summary>
+        /// Applies 'decimal' logical type validation for a given logical schema.
+        /// </summary>
+        /// <param name="schema">The schema to be validated.</param>
+        public override void ValidateSchema(LogicalSchema schema)
+        {
+            if (Schema.Type.Bytes != schema.BaseSchema.Tag && Schema.Type.Fixed != schema.BaseSchema.Tag)
+                throw new AvroTypeException("'decimal' can only be used with an underlying bytes or fixed type");
+
+            var precisionVal = schema.GetProperty("precision");
+
+            if (string.IsNullOrEmpty(precisionVal))
+                throw new AvroTypeException("'decimal' requires a 'precision' property");
+
+            var precision = int.Parse(precisionVal, CultureInfo.CurrentCulture);
+
+            if (precision <= 0)
+                throw new AvroTypeException("'decimal' requires a 'precision' property that is greater than zero");
+
+            var scale = GetIntPropertyValueFromSchema(schema, "scale");
+
+            if (scale < 0 || scale > precision)
+                throw new AvroTypeException("'decimal' requires a 'scale' property that is zero or less than or equal to 'precision'");
+        }
+
+        /// <summary>
+        /// Converts a logical value to an instance of its base type.
+        /// </summary>
+        /// <param name="logicalValue">The logical value to convert.</param>
+        /// <param name="schema">The schema that represents the target of the conversion.</param>
+        /// <returns>An object representing the encoded value of the base type.</returns>        
+        public override object ConvertToBaseValue(object logicalValue, LogicalSchema schema)
```

Comment:
Rather than copy-pasting the documentation of each of these methods from `LogicalType`, use `/// <inheritdoc/>`. Only copy-paste-modify if the specific override of the method requires some customizations.

Reply:
I didn't know about this - thanks!